pilot/pkg/xds/proxy_dependencies.go

```diff
@@ -67,6 +68,13 @@ func checkProxyDependencies(proxy *model.Proxy, config model.ConfigKey, push *mo
 		}
 	case model.Router:
 		if config.Kind == kind.ServiceEntry {
+			if features.FilterGatewayClusterConfig {
+				hostsFromGateways := push.GetHostsFromGateways(proxy)
```

Comment:
this may be called many times, any concern about cost to compute? I suspect it's fine

Reply:
Yeah I have the same concern, I've add a cache for it